File: WebApp/src/app/shared/models/document/OpsTransaction.model.ts

import { Container } from "./container.model";
import { PermissionShipment } from "./permissionShipment";
import { BaseModel } from "../base.model";


export class OpsTransaction extends BaseModel {
    userModified: string = null;
    userCreated: string = null;
    sumPackages: number = null;
    sumContainers: number = null;
    sumCbm: number = null;
    sumChargeWeight: number = null;
    sumGrossWeight: number = null;
    sumNetWeight: number = null;
    fieldOpsId: string = null;
    invoiceNo: string = null;
    warehouseId: string = null;
    finishDate: any = null;
    billingOpsId: string = null;
    purchaseOrderNo: string = null;
    flightVessel: string = null;
    supplierId: string = null;
    pod: string = null;
    pol: string = null;
    clearanceLocation: string = null;
    customerId: string = null;
    customerName: string = null;
    shipmentMode: string = null;
    serviceMode: string = null;
    productService: string = null;
    serviceDate: any = null;
    hwbno: string = null;
    mblno: string = null;
    jobNo: string = null;
    id: string = "00000000-0000-0000-0000-000000000000";
    hblid: string = "00000000-0000-0000-0000-000000000000";
    agentId: string = null;
    salemanId: string = null;
    supplierName: string = null;
    agentName: string = null;
    podName: string = null;
    polName: string = null;
    packageTypeId: number = null;
    csMawbcontainers: Container[] = null;
    containerDescription: string = '';
    commodityGroupId: string = '';
    isLocked: boolean = false;
    userCreatedName: string = '';
    userModifiedName: string = '';
    currentStatus: string = '';
    shipper: string = '';
    consignee: string = '';
    shipmentType: string = null;
    serviceNo: string = null;
    serviceHblId: string = null;
    linkSource: string = null;
    replicatedId: string = null;

    note: string = '';
    isLinkFee: boolean = false;
    isLinkJob: boolean = false;

    permission: PermissionShipment = new PermissionShipment();

    // * Custom
    isReplicate: boolean = false;
    isAllowChangeSaleman: boolean = true;
    salesmanName: string = null;

    constructor(object?: any) {
        super();
        const self = this;
        for (const key in object) {
            if (self.hasOwnProperty(key.toString())) {
                self[key] = object[key];
            }
        }
    }
}

File: WebAPI/eFMS.API.SystemWeb/eFMS.API.Documentation/eFMS.API.Documentation.DL/Models/OpsTransactionModel.cs
﻿using eFMS.API.Common.Models;
using eFMS.API.Documentation.Service.Models;
using System.Collections.Generic;

namespace eFMS.API.Documentation.DL.Models
{
    public class OpsTransactionModel : OpsTransaction
    {
        public List<CsMawbcontainerModel> CsMawbcontainers { get; set; }
        public string PODName { get; set; }
        public string POLName { get; set; }
        public int CurrentStageId { get; set; }
        public string CurentStageCode { get; set; }
        public string AgentName { get; set; }
        public string SupplierName { get; set; }
        public string ClearanceNo { get; set; }
        public string CustomerName { get; set; }
        public PermissionAllowBase Permission { get; set; }
        public string UserCreatedName { get; set; }
        public string UserModifiedName { get; set; }
        public string ProductService { get; set; }
        public bool IsLinkJob { get; set; }
        public bool IsReplicate { get; set; }
        public string ReplicateJobNo { get; set; }
        public string UserCreatedNameLinkJob { get; set; }
        public bool IsAllowChangeSaleman { get; set; }
        public string SalesmanName { get; set; }
        public string DepartmentName { get; set; }
        public string GroupName { get; set; }
    }
}
